### PR TITLE
transform: Compile releases - if releases with no dates, compile rele…

### PR DESCRIPTION
**Pull request by odscjames**
…ases without dates and warn

https://github.com/open-contracting/kingfisher-process/issues/279

### CODE REVIEW

**Comment by odscjames** [comment]
[Compile Release Transform: date missing](https://trello.com/c/sRtiAXxV/205-compile-release-transform-date-missing)

File: ocdskingfisherprocess/transform/compile_releases.py [hunk 1]
```diff
@@ -113,22 +113,26 @@ def _process_record(self, ocid, record, warnings=None):
 
         releases = record.get('releases', [])
         releases_linked = [r for r in releases if is_linked_release(r)]
```
[the remark this answers is not in the document]
Comment (odscjames) [reply ocdskingfisherprocess/transform/compile_releases.py:115]: I'm not sure that is the logic we want; previously we had said that if any releases are linked we shouldn't try and do things with releases? But we can change of course:

So the situation is we have a record with some releases linked, and some non-linked releases with dates:
A) we ignore all releases and carry on with next steps (look for compiledRelease, release tagged compiled)
B) we compile ourselves with the non-linked releases, and include a warning/note/etc so that analysts know some data was not included in this

[the remark this answers is not in the document]
Comment (jpmckinney) [reply ocdskingfisherprocess/transform/compile_releases.py:115]: The earlier discussion hadn't accounted for releases without dates. The new logic is:

1. Partition the releases into those with and without dates.
1. Warn if there are any without dates.
1. Of the releases with dates, if none are linked, then we can merge them.
1. Otherwise, we need to look for alternatives as before.

Linked releases are only a problem because we don't want to send requests to collect them remotely (it's slow, the URLs might no longer exist, etc.). However, if they don't have dates, then we can't use them, and we'll ignore them – so we don't encounter the previous problem.

[the remark this answers is not in the document]
Comment (odscjames) [reply ocdskingfisherprocess/transform/compile_releases.py]: Done. This is a very edge case, so I think the existing test's I've added cover enough cases given the incoming rewrite.
